# Subject: Classical Data Analysis

## Session 1 - Regression

### Exercise 2 


Implementation of Python StatsModels package with Quandl integration to calculate the Linear regression with one variable.  

Considering the OLS presented in Demo 3 (Outbound tourism statistics for Spain), develop a new regression analysis based on the Quandl dataset “Inbound tourism statistics for Spain”. This dataset is available at https://www.quandl.com/data/UTOR/INTUR_ESP

Use the field “Tourist arrivals at national borders – Thousands” as independent variable and the field “Travel - US$ Mn” as dependent variable.

- Interpret and discuss the OLS Regression Results. 
- Commit scripts in your GitHub account. You should export your solution code (.ipynb notebook) and push it to your repository “ClassicalDataAnalysis”.

The following are the tasks that should complete and synchronize with your repository “ClassicalDataAnalysis” until October 13. Please notice that none of these tasks is graded, however it’s important that you correctly understand and complete them in order to be sure that you won’t have problems with further assignments.


## Import Quandl data using your API Key

In [5]:
import quandl
exercise = quandl.get("UTOR/EXTUR_ESP", authtoken="mqxBUtsyqyhCCXQQxNNi")
exercise.head()

Departures - Thousands  \
Date                                 
1995-12-31                  3648.0   
1996-12-31                  4224.0   
1997-12-31                  3980.0   
1998-12-31                  4794.0   
1999-12-31                  3519.0   

            Tourism expenditure in other countries - US$ Mn  Travel - US$ Mn  \
Date                                                                           
1995-12-31                                           5826.0           4539.0   
1996-12-31                                           6332.0           5057.0   
1997-12-31                                           6163.0           4681.0   
1998-12-31                                           7119.0           5288.0   
1999-12-31                                           7803.0           5862.0   

            Passenger transport - US$ Mn  
Date                                      
1995-12-31                        1287.0  
1996-12-31                        1275.0  
1997-12-31                        1482.0  
1998-12-31                        1831.0  
1999-12-31                        1941.0

In [16]:
import pandas as pd
exercise_control=pd.read_csv("C:/Users/Ecem/Desktop/BTS/Classical_Data_Analysis/UTOR-INTUR_ESP.csv")
exercise_control.head()

Date  Tourist arrivals at national borders - Thousands  \
0  2014-12-31                                           64995.0   
1  2013-12-31                                           60675.0   
2  2012-12-31                                           57464.0   
3  2011-12-31                                           56177.0   
4  2010-12-31                                           52677.0   

   Tourism expenditure in the country - US$ Mn  Travel - US$ Mn  \
0                                          NaN          65100.0   
1                                          NaN          62584.0   
2                                          NaN          57877.0   
3                                          NaN          62447.0   
4                                          NaN          54305.0   

   Passenger transport - US$ Mn  
0                           NaN  
1                           NaN  
2                           NaN  
3                           NaN  
4                           NaN

In [ ]:
2####the same meaning 'Tourist arrivals at national borders - Thousands' and  'Departures - Thousands' probably only differrence filter of date

## Regression model with Statsmodels and without a constant

In [8]:
import statsmodels.api as sm

In [9]:
X = exercise["Departures - Thousands"]
y = exercise["Travel - US$ Mn"]

In [10]:
model = sm.OLS(y, X).fit()

In [11]:
predictions = model.predict(X) 

In [12]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        Travel - US$ Mn   R-squared:                       0.976
Model:                            OLS   Adj. R-squared:                  0.975
Method:                 Least Squares   F-statistic:                     784.6
Date:                Fri, 13 Oct 2017   Prob (F-statistic):           6.48e-17
Time:                        09:36:05   Log-Likelihood:                -180.83
No. Observations:                  20   AIC:                             363.7
Df Residuals:                      19   BIC:                             364.7
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Departures - Thousands     1.4936      0.053     28.011      0.000       1.382       1.605
==============================================================================
Omnibus:                        1.502   Durbin-Watson:                   0.912
Prob(Omnibus):                  0.472   Jarque-Bera (JB):                1.247
Skew:                           0.560   Prob(JB):                        0.536
Kurtosis:                       2.510   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Interpreting the Table 

there is a 95% confidence; "Tourist arrivals at national borders - Thousands" variable increases by 1, the predicted value of "Travel - US$ Mn" increases by 1.4936. R-Square Means:Using "Tourist arrivals at national borders - Thousands" variables, the percentage of explain model,0.976

## Regression model with Statsmodels and with a constant

In [14]:
import statsmodels.api as sm

X = exercise["Departures - Thousands"]
y = exercise["Travel - US$ Mn"]

In [15]:
X = sm.add_constant(X) 
modelwc = sm.OLS(y, X).fit()
predictions = modelwc.predict(X)
modelwc.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        Travel - US$ Mn   R-squared:                       0.877
Model:                            OLS   Adj. R-squared:                  0.870
Method:                 Least Squares   F-statistic:                     128.4
Date:                Fri, 13 Oct 2017   Prob (F-statistic):           1.26e-09
Time:                        09:38:05   Log-Likelihood:                -180.42
No. Observations:                  20   AIC:                             364.8
Df Residuals:                      18   BIC:                             366.8
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==========================================================================================
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
const                    939.2279   1084.358      0.866      0.398   -1338.923    3217.379
Departures - Thousands     1.3975      0.123     11.332      0.000       1.138       1.657
==============================================================================
Omnibus:                        2.126   Durbin-Watson:                   0.871
Prob(Omnibus):                  0.345   Jarque-Bera (JB):                1.741
Skew:                           0.673   Prob(JB):                        0.419
Kurtosis:                       2.475   Cond. No.                     2.02e+04
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.02e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### Interpreting the Table 

there is a 95% confidence;
"Tourist arrivals at national borders - Thousands" variable increases by 1, the predicted value of "Travel - US$ Mn" increases by 1.3975 and y-intercept 939.2279.
R-Square Means:Using "Tourist arrivals at national borders - Thousands" variables, the percentage of explain model,0.877